lti_consumer/plugin/views.py

```diff
@@ -130,3 +144,84 @@ def get_queryset(self):
     def perform_create(self, serializer):
         lti_configuration = self.request.lti_configuration
         serializer.save(lti_configuration=lti_configuration)
+
+    @action(
+        detail=True,
+        methods=['GET'],
+        renderer_classes=[LineItemResultsRenderer]
+    )
+    def results(self, request, *args, **kwargs):
+        """
+        Return a Result list for an LtiAgsLineItem
+
+        Query Parameters:
+          * user_id (string): String external user id representation. Only
+                one record per unique `user_id` will be returned
+          * limit (integer): The maximum number of records to return. Records are
+                sorted with most recent timestamp first
+
+        Returns:
+          * An array of Result records, formatted by LtiAgsResultSerializer
+                and returned with the media-type for LineItemResultsRenderer
+        """
+        line_item = self.get_object()
+        scores = line_item.scores.all().order_by('-timestamp')
+
+        if request.query_params.get('user_id'):
+            scores = scores.filter(user_id=request.query_params.get('user_id'))
+
+        # Only get one record per user
+        # Would use scores.distinct('user_id') here, but MySQL does
+        # not support it distinct with fields
+        results = {}
+        for score in scores:
+            if score.user_id not in results:
+                results[score.user_id] = score
+        results = results.values()
+
+        if request.query_params.get('limit'):
+            results = results[:int(request.query_params.get('limit'))]
+
+        serializer = LtiAgsResultSerializer(
+            list(results),
+            context={'request': self.request},
+            many=True,
+        )
+
+        return Response(serializer.data)
+
+    @action(
+        detail=True,
+        methods=['POST'],
+        parser_classes=[LineItemScoreParser],
+        renderer_classes=[LineItemScoreRenderer]
+    )
+    def scores(self, request, *args, **kwargs):
+        """
+        Create a Score record for an LtiAgsLineItem
+
+        Data:
+          * A JSON object capable of being serialized by LtiAgsScoreSerializer
+
+        Returns:
+          * An copy of the saved record, formatted by LtiAgsScoreSerializer
+                and returned with the media-type for LineItemScoreRenderer
+        """
+        line_item = self.get_object()
+
+        user_id = request.data.get('userId')
+        existing_score = line_item.scores.filter(user_id=user_id).first()
```

Comment:
@pcockwell You can use `line_item.scores.get(user_id=user_id)`, given the unique constraint.

Reply:
I'll add a comment to make it explicit, but I want `existing_score` to be set to `None` if there are no scores. Using `filter` and `first` does that, whereas `get` raises a `DoesNotExist` error.